code/game/mecha/mech_fabricator.dm

```diff
@@ -84,19 +84,16 @@
 	return 1
 
 /obj/machinery/mecha_part_fabricator/emag_act()
-	if(emagged)

```

Comment:
I honestly prefer this  the old way. Just make the `emagged = FALSE` be a `TRUE` and remove the last line.